torch/cuda/_utils.py

```diff
@@ -159,7 +208,7 @@ def check_nvrtc(result: int) -> None:
     check_nvrtc(libnvrtc.nvrtcGetPTX(prog, ptx))
     libnvrtc.nvrtcDestroyProgram(ctypes.byref(prog))
 
-    return ptx.value
+    return ptx.raw if torch.version.hip else ptx.value
```

Comment:
This was done because hipRTC generates raw CO binaries instead of PTX, and for some reason `.value` causes the string to be truncated, likely due to the presence of `\0` in the string. So we use `.raw` instead.